src/Microsoft.IdentityModel.Protocols.WsFederation/WsFederationMetadataSerializer.cs

```diff
@@ -248,6 +258,75 @@ protected virtual string ReadPassiveRequestorEndpoint(XmlReader reader)
             return tokenEndpoint;
         }
 
+        /// <summary>
+        /// Read fed:SecurityTokenServiceEndpoint element from metadata XML.
+        /// </summary>
+        /// <param name="reader"><see cref="XmlReader"/> used to read SecurityTokenServiceEndpoint.</param>
+        /// <returns>Active token endpoint string</returns>
+        /// <exception cref="XmlReadException">If an error occurs while reading the SecurityTokenServiceEndpoint</exception>
+        protected virtual string ReadSecurityTokenServiceEndpoint(XmlReader reader)
+        {
+            XmlUtil.CheckReaderOnEntry(reader, Elements.SecurityTokenServiceEndpoint, Namespace);
+
+            // <SecurityTokenServiceEndpoint>
+            if (reader.IsEmptyElement)
+                throw XmlUtil.LogReadException(LogMessages.IDX22812, Elements.SecurityTokenServiceEndpoint);
+
+            reader.ReadStartElement();
+            reader.MoveToContent();
+
+            // <EndpointReference>
+            XmlUtil.CheckReaderOnEntry(reader, WsAddressing.Elements.EndpointReference, WsAddressing.Namespace);
+            if (reader.IsEmptyElement)
+                throw XmlUtil.LogReadException(LogMessages.IDX22812, WsAddressing.Elements.EndpointReference);
+
+            reader.ReadStartElement(WsAddressing.Elements.EndpointReference, WsAddressing.Namespace);
+            reader.MoveToContent();
+
+            if (reader.IsEmptyElement)
+                throw XmlUtil.LogReadException(LogMessages.IDX22814);
+
+            string tokenEndpoint = null;
+
+            while (reader.IsStartElement())
+            {
+                if (reader.IsStartElement(WsAddressing.Elements.Address, WsAddressing.Namespace))
+                {
+                    // <Address>
+                    XmlUtil.CheckReaderOnEntry(reader, WsAddressing.Elements.Address, WsAddressing.Namespace);
+
+                    if (reader.IsEmptyElement)
+                        throw XmlUtil.LogReadException(LogMessages.IDX22812, WsAddressing.Elements.Address);
+
+                    reader.ReadStartElement(WsAddressing.Elements.Address, WsAddressing.Namespace);
+                    reader.MoveToContent();
+
+                    tokenEndpoint = Trim(reader.ReadContentAsString());
+
+                    if (string.IsNullOrEmpty(tokenEndpoint))
+                        throw XmlUtil.LogReadException(LogMessages.IDX22814);
+
+                    // </Address>
+                    reader.MoveToContent();
+                    reader.ReadEndElement();
+                }
+                else
+                {
+                    reader.ReadOuterXml();
+                }
+            }
+
+            // </EndpointReference>
+            reader.MoveToContent();
+            reader.ReadEndElement();
+
+            // </PassiveRequestorEndpoint>
```

Comment:
Should this comment be: // </SecurityTokenServiceEndpoint>

Reply:
Yes, updating

Reply:
Looks like this was marked resolved but also not updated in the branch so far as I can tell. @victorm-hernandez do you have local changes which still will be persisted?

Reply:
You are right, to avoid any confusion I will mark elements as resolved once I sent another iteration.